packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -351,22 +351,21 @@ typedef TweenVisitor<T extends Object> = Tween<T>? Function(Tween<T>? tween, T t
 abstract class ImplicitlyAnimatedWidgetState<T extends ImplicitlyAnimatedWidget> extends State<T> with SingleTickerProviderStateMixin<T> {
   /// The animation controller driving this widget's implicit animations.
   @protected
-  AnimationController? get controller => _controller;
-  AnimationController? _controller;
+  AnimationController get controller => _controller;
+  late final AnimationController _controller = AnimationController(
+      duration: widget.duration,
+      debugLabel: kDebugMode ? widget.toStringShort() : null,
+      vsync: this,
+    );
```

Comment:
This lazily initializes `AnimationController` whenever _controller is first accessed. 

I think the convention here is to define the `AnimationController` in initState. Shouldn't we still keep it that way since we aren't guaranteed to have access to the `StatefulWidget` before `_controller` is called?

Reply:
Also, nit if we keep this as-is:
```suggestion
  late final AnimationController _controller = AnimationController(
    duration: widget.duration,
    debugLabel: kDebugMode ? widget.toStringShort() : null,
    vsync: this,
  );
```

Reply:
My reasoning to use the initializers is that it gives a guarantee through Dart's language design (rather than Flutter's State class design and usage conventions) that the objects will be initialized when accessed, whereas initializing them in `initState` does not.

The downside to this is that unused values will be initialized by the `dispose` call even if they where otherwise not used (i.e. `initState` was never called). Though I assumed that either never happens or is rare enough that it doesn't matter.

Reply:
Lazily initializing this here seems fine to me.

Can you fix the formatting though?